src/ball_tree.jl

```diff
@@ -3,9 +3,9 @@
 # which radius are determined from the given metric.
 # The tree uses the triangle inequality to prune the search space
 # when finding the neighbors to a point,
-immutable BallTree{T <: AbstractFloat, M <: Metric} <: NNTree{T, M}
-    data::Matrix{T}                       # dim x n_points array with floats
-    hyper_spheres::Vector{HyperSphere{T}} # Each hyper sphere bounds its children
+immutable BallTree{V <: AbstractVector, N, T, M <: Metric} <: NNTree{V, M}
+    data::Vector{V}
```

Comment:
For `isbits(T)` yes. However, `Matrix{T}` with `isbits(T)` will be reinterpreted to a vector of static vectors so for any previous usage of this package the memory layout will be the same. I haven't written documentation yet but the requirement on `V` is that it defined both length and eltype which pretty much excludes anything other than fixed size arrays.